ca_gazebo/src/create_virtual_wall_plugin.cpp

```diff
@@ -124,21 +128,31 @@ void VirtualWallSensorPlugin::OnUpdate()
         sensor_pose.Pos().Y(),
         sensor_pose.Pos().Z());
 
-    // distance from P to straight line X=X0+lambda*V is:
-    // sin(phi)*abs(P-X0) where cos(phi)=(P-X0)*V/abs(P-X0)
+    // https://www.qc.edu.hk/math/Advanced%20Level/Point_to_line.htm
+    // Pythagoras Theorem
+    // Get distance from the RCON (robot's omni sensor) to the VW sensor
     const auto & error = (sensor_pose.Pos() - laser_pose.Pos());
-    double phi = std::acos(error.Dot(laser_dir) / error.Length());
-    double d = error.Length() * std::sin(phi);
-    double lambda = (sensor_pose.Pos().X() - laser_pose.Pos().X()) / laser_dir.X();
-    ROS_DEBUG_NAMED("virtual_wall_plugin", "d=%f lambda=%f detected_range=%f error=%f",
+    // Get the projected error into the line generated by the VW sensor and its laser
+    const double proj_error = error.Dot(laser_dir);
+    // Get the distance from the robot to the line generated by the VW sensor and its laser
+    // by doing a simple vector difference
+    const double proj_error_sq = proj_error * proj_error;
+    const double error_sq = error.Length() * error.Length();
+    const double d = std::sqrt(error_sq - proj_error_sq);
+    ROS_DEBUG_NAMED("virtual_wall_plugin", "d=%f detected_range=%f error=%f",
         d,
-        lambda,
         detected_range,
         error.Length());
 
     std_msgs::Bool msg;
-    bool in_range = detected_range > (error.Length() - 0.1) && error.Length() < this->sensor_->RangeMax();
-    msg.data = d < 0.1 && lambda > 0 && in_range;
+    // The line constituted by VW laser has the from y = m*x + b
+    // So to check if the robot is in the x < 0:
+    // x = (y - b) / m, where 'y' is the component of the robot
+    const bool in_front_of_laser = ((sensor_pose.Pos().Y() - laser_pose.Pos().Y()) / (laser_dir.Y()/laser_dir.X())) > 0;
```

Comment:
I think this part is the one that is causing trouble because it isn't taking to account the rotation, only working with fixed axis.